Assets/Mirror/Runtime/Transport/LLAPITransport.cs

```diff
@@ -141,9 +141,7 @@ public bool ProcessClientMessage()
                     OnClientConnected.Invoke();
                     break;
                 case NetworkEventType.DataEvent:
-                    byte[] data = new byte[receivedSize];
-                    Array.Copy(clientReceiveBuffer, data, receivedSize);
-                    OnClientDataReceived.Invoke(data);
+                    OnClientDataReceivedNonAlloc.Invoke(new ArraySegment<byte>(clientReceiveBuffer, 0, receivedSize));
```

Comment:
This is nice.   Note this is possible only with transports that don't queue up the messages.